components/raftstore/src/store/snapshot_backup.rs

```diff
@@ -0,0 +1,388 @@
+            let res = self
+                .before
+                .fetch_update(Ordering::SeqCst, Ordering::SeqCst, |v| {
+                    if v > new_lease { None } else { Some(new_lease) }
+                });
+            match res {
+                Ok(_) => {
+                    metrics::SNAP_BR_SUSPEND_COMMAND_LEASE_UNTIL.set(new_lease as _);
+                    break;
+                }
+                Err(prev) => v = prev,
+            }
+        }
+        if last_lease_valid {
+            metrics::SNAP_BR_LEASE_EVENT.renew.inc();
+        } else {
+            metrics::SNAP_BR_LEASE_EVENT.create.inc();
+        }
+        last_lease_valid
+    }
+
+    pub fn reset(&self) {
+        self.before.store(0, Ordering::SeqCst);
+        metrics::SNAP_BR_SUSPEND_COMMAND_LEASE_UNTIL.set(0);
+        metrics::SNAP_BR_LEASE_EVENT.reset.inc();
+    }
+}
+
+impl Coprocessor for Arc<PrepareDiskSnapObserver> {
+    fn start(&self) {
+        self.initialized.store(true, Ordering::Release)
+    }
+
+    fn stop(&self) {
+        self.initialized.store(false, Ordering::Release)
+    }
+}
+
+impl QueryObserver for Arc<PrepareDiskSnapObserver> {
+    fn pre_propose_query(
+        &self,
+        cx: &mut crate::coprocessor::ObserverContext<'_>,
+        reqs: &mut Vec<kvproto::raft_cmdpb::Request>,
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        for req in reqs {
+            if req.has_ingest_sst() {
+                // Note: this will reject the batch of commands, which isn't so effective.
+                // But we cannot reject proposing a subset of command for now...
+                cx.bypass = true;
+                metrics::SNAP_BR_SUSPEND_COMMAND_TYPE
+                    .with_label_values(&["Ingest"])
+                    .inc();
+                return Err(self.reject());
+            }
+        }
+        Ok(())
+    }
+}
+
+impl AdminObserver for Arc<PrepareDiskSnapObserver> {
+    fn pre_propose_admin(
+        &self,
+        _: &mut crate::coprocessor::ObserverContext<'_>,
+        admin: &mut kvproto::raft_cmdpb::AdminRequest,
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        if matches!(
+            admin.get_cmd_type(),
+            AdminCmdType::Split |
+            AdminCmdType::BatchSplit |
+            // We disable `CompactLog` here because if the log get truncated, 
+            // we may take a long time to send snapshots during restoring.
+            AdminCmdType::CompactLog |
+            // We will allow `Commit/RollbackMerge` here because the 
+            // `wait_pending_admin` will wait until the merge get finished.
+            // If we reject them, they won't be able to see the merge get finished.
+            // And will finally time out.
+            AdminCmdType::PrepareMerge |
+            AdminCmdType::ChangePeer |
+            AdminCmdType::ChangePeerV2
```

Comment:
How about `BatchSwitchWitness`?

Reply:
I think they won't make things worse. However we can still add it since rejecting it won't be harmful anyway.